create-i18n/index.test.ts

```diff
@@ -233,4 +236,237 @@ test('tracks double definition', () => {
+  equal(i18n.loading.get(), false)
+  equal(partialGetCalls, [])
+
+  postUnbind = post.subscribe(t => {
+    events.push(t.title)
+  })
+  headingUnbind = heading.subscribe(t => {
+    events.push(t.title)
+  })
+  equal(i18n.loading.get(), true)
+  equal(partialGetCalls, [{ ru: ['main/post'] }])
+  equal(events, ['Post', 'Title'])
+})
+
+test('loads translations partial', async () => {
+  await partialGetResponse([
+    {
+      'main/post': { title: 'Публикация' },
+      'main/heading': { title: 'Заголовок' },
+      'main/comment': { title: 'Комментарий' }
+    }
+  ])
+  equal(i18n.loading.get(), false)
+  equal(events, ['Post', 'Title', 'Публикация', 'Заголовок'])
+  equal(post.get(), { title: 'Публикация' })
+  equal(heading.get(), { title: 'Заголовок' })
+  equal(comment.get(), { title: 'Комментарий' })
+  events = []
+})
+
+test('after components mount should load translations with different prefix or without prefix names', async () => {
+  equal(i18n.loading.get(), false)
+  equal(partialGetCalls, [])
+
+  let userUnbind = user.subscribe(t => {
+    events.push(t.title)
+  })
+  let gamesUnbind = games.subscribe(t => {
+    events.push(t.title)
+  })
+  equal(i18n.loading.get(), true)
+  equal(partialGetCalls, [{ ru: ['settings/user'] }, { ru: ['games'] }])
+  equal(events, ['User', 'Games'])
+
+  await partialGetResponse([
+    {
+      'settings/user': { title: 'Пользователь' },
+      'games': { title: 'Игры' }
+    }
+  ])
+  equal(i18n.loading.get(), false)
+  equal(events, [
+    'User',
+    'Games',
+    'Публикация',
+    'Заголовок',
+    'Пользователь',
+    'Игры'
+  ])
+  userUnbind()
+  gamesUnbind()
+  events = []
+})
+
+test("component mounting shouldn't load cached translations", async () => {
+  commentUnbind = comment.subscribe(t => {
+    events.push(t.title)
+  })
+  equal(i18n.loading.get(), false)
+  equal(partialGetCalls, [])
+  equal(events, ['Комментарий'])
+  events = []
+})
+
+test('loads translation after component mounted', async () => {
+  message.subscribe(t => {
+    events.push(t.title)
+  })
+  equal(i18n.loading.get(), true)
+  equal(partialGetCalls, [{ ru: ['chat/message'] }])
+  equal(events, ['Message'])
+
+  await partialGetResponse([
+    {
+      'chat/message': { title: 'Сообщение' }
+    }
+  ])
+  equal(i18n.loading.get(), false)
+  equal(events, [
+    'Message',
+    'Публикация',
+    'Заголовок',
+    'Комментарий',
+    'Сообщение'
+  ])
+  equal(message.get(), { title: 'Сообщение' })
+  events = []
+})
+
+test("locale changing shouldn't load cached translations", async () => {
+  locale.set('en')
+  equal(i18n.loading.get(), false)
+  equal(partialGetCalls, [])
+  equal(events, ['Post', 'Title', 'Comment', 'Message'])
+  events = []
+})
+
+test('locale changing should load translations for mounted components only', async () => {
+  postUnbind()
+  headingUnbind()
+  commentUnbind()
+
+  // 1s delay for components unmounting
+  await new Promise(resolve => setTimeout(resolve, 1100))
```

Comment:
1. `nanostores` exports constant with this delay
2. Let’s use `nanodelay` here

Reply:
done